app/models/vm_or_template.rb

```diff
@@ -1061,6 +1061,10 @@ def miq_server_proxies
     when 'VMware'
       # VM cannot be scanned by server if they are on a repository
       return [] if storage_id.blank? || self.repository_vm?
+    when 'RedHat'
+      return [] if storage_id.blank?
+    when 'Microsoft'
```

Comment:
Type-specific versions of `miq_server_proxies` and `proxies4job` should be implemented in `microsoft/infra_manager/vm_or_template_shared/scanning.rb` - eliminating the need for type-specific checks in the common code.